### ISSUE TITLE
Django `main` – `models.E025`: The property 'revisions'/'workflow_states' clashes with a related field accessor

### ISSUE THREAD

**Issue by laymonage** [wagtail/wagtail#11746]
### Issue Summary

Tests are failing against Django's `main` branch with the following system check errors:

```
SystemCheckError: System check identified some issues:

ERRORS:
tests.FullFeaturedSnippet: (models.E025) The property 'revisions' clashes with a related field accessor.
tests.FullFeaturedSnippet: (models.E025) The property 'workflow_states' clashes with a related field accessor.
```

### Steps to Reproduce

1. Install Django from its `main` branch
2. Run Wagtail's tests

### Technical details

The error shows up as of django/django#17936.

Per discussion in https://github.com/wagtail/wagtail/pull/11591#pullrequestreview-1863244775 we ought to implement the default `revisions` and `workflow_states` in the mixins as `GenericRelation`s anyway, so perhaps we can tackle the two things at once.

**Comment by laymonage** [comment]
This might be a bug in Django itself, I've raised https://code.djangoproject.com/ticket/35301 in upstream.

**Comment by laymonage** [comment]
This has been fixed in upstream, but I'd still like to get #11750 in for 6.1.
